Sources/Utilities/KMZendeskChatHandler.swift

```diff
@@ -0,0 +1,238 @@
+    
+    public func initiateZendesk(key: String, conversationId: String) {
+        Chat.initialize(accountKey: key)
+        groupId = conversationId
+        zenChatIntialized = true
+        fetchUserId()
+        authenticateUser()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
+            self.sendChatInfo()
+            self.sendChatTranscript()
+            
+            let connectionToken = Chat.connectionProvider?.observeConnectionStatus { [self] (connection) in
+                // Handle connection status changes
+                guard connection.isConnected && connectionStatus != true else {
+                   self.connectToZendeskSocket()
+                   return
+                }
+                connectionStatus = true
+                observeChatLogs()
+                if !isChatTranscriptSent {
+                    sendChatTranscript()
+                }
+               
+           }
+           
+       }
+    }
+    
+    func fetchUserId() {
+        guard let id = ALUserDefaultsHandler.getUserId() else {return}
+        self.userId = id
+    }
+    
+    public func getGroupId() -> String {
+        return groupId
+    }
+    func authenticateUser() {
+        let chatAPIConfiguration = ChatAPIConfiguration()
+        chatAPIConfiguration.visitorInfo = VisitorInfo(name: ALUserDefaultsHandler.getDisplayName() ?? "", email: ALUserDefaultsHandler.getEmailId() ?? "", phoneNumber: "")
+        Chat.instance?.configuration = chatAPIConfiguration
+        connectToZendeskSocket()
+    }
+    var chatLogToken : ObservationToken?
+    
+    func observeChatLogs() {
+        guard Chat.connectionProvider?.status == .connected else {
+            connectToZendeskSocket()
+            return
+        }
+        
+        let stateToken = Chat.chatProvider?.observeChatState { (state) in
+            // Handle logs, agent events, queue position changes and other events
+        }
+        chatLogToken = stateToken
+    }
+    
+    func connectToZendeskSocket() {
+        Chat.connectionProvider?.connect()
+    }
+    
+    func sendMessage(message: ALMessage) {
+        guard Chat.connectionProvider?.status == .connected else {
+            connectToZendeskSocket()
+            return
+        }
+        
+        sendMessageToZendesk(message: message.message) { (result) in
+            switch result {
+            case .success:
+                print("Message Sent to Zendesk Successfully")
+            case .failure(let error):
+                print("Failed Send the Message to Zendesk \(error)")
+            }
+        }
+    }
+    
+    func sendMessageToZendesk(message: String, completionHandler: @escaping (Result<String, ChatProvidersSDK.DeliveryStatusError>) -> Void) {
+        Chat.chatProvider?.sendMessage(message) { (result) in
+            completionHandler(result)
+        }
+    }
+    
+    func sendChatTranscript() {
+        let chanelKey = NSNumber(value: Int(groupId) ?? 0)
```

Comment:
Instead of global var why don't we pass the group Id? It's being called from just one function where group ID is present.